system/datatypes/mailinglist.php

```diff
@@ -421,78 +421,80 @@ private function editMailingAction($mailingId = null) {
 		// throw error if edit is requested without client logged in
 		if (!isUser()) return notify('error', __('login-to-edit'));
 
+		$sendMailing = $this->isPosted(self::FORM_CMD_MAILING_SEND);
+		$saveMailing = $this->isPosted(self::FORM_CMD_MAILING_SAVE);
+		if ($this->isPosted() && !$sendMailing && !$saveMailing) {
+			return notify('error', __('ml_invalid-command'));
+		}
+		$isPosted = $sendMailing || $saveMailing;
+
 		// get mailing if id was given
 		/** @var HyphaDomElement $mailing */
-		if ($mailingId) {
+		if (null != $mailingId) {
 			// check if given id was correct
 			$mailing = $this->xml->document()->getElementById($mailingId);
 			if (!$mailing instanceof HyphaDomElement) {
-				return notify('error', __('invalid-id-given'));
+				return notify('error', __('not-found')); // 404
 			}
 			// check if status is correct
-			if ($mailing->getAttribute('status') == self::MAILING_STATUS_SENT) {
+			$status = $mailing->getAttribute('status');
+			if (self::MAILING_STATUS_SENT == $status && $sendMailing) {
+				notify('success', __('ml_successfully_sent'));
```

Comment:
This is another case where you either say "Mail was sent" or "Mail was already sent". Since sending a mail is actually an action, not just a state change, and you would rather not send duplicate mails, it seems more sensible to give more accurate feedback to the user. If a user somehow accidentally tries to send the mailing twice, they might think they sent the mailing list twice if the result is just "successfully sent". Alternately, if the sending fails (or at least the user thinks so) and they attempt to send it again (e.g. from a browser window they still had open from before it was originally sent), they might think the message was sent (again), when in reality nothing happened.